src/concurrency/thread.rs

```diff
@@ -147,6 +153,28 @@ impl<'mir, 'tcx> Thread<'mir, 'tcx> {
     fn thread_name(&self) -> &[u8] {
         if let Some(ref thread_name) = self.thread_name { thread_name } else { b"<unnamed>" }
     }
+
+    fn get_top_user_relevant_frame(&self) -> Option<usize> {
```

Comment:
This sounds like a simple getter but is an expensive computation -- I think it should be named differently. Maybe `compute_`?